src/ripple/shamap/impl/SHAMapDelta.cpp

```diff
@@ -123,12 +123,12 @@ SHAMap::compare (std::shared_ptr<SHAMap> const& otherMap,
 
     assert (isValid () && otherMap && otherMap->isValid ());
 
-    using StackEntry = std::pair <SHAMapTreeNode*, SHAMapTreeNode*>;
-    std::stack <StackEntry, std::vector<StackEntry>> nodeStack; // track nodes we've pushed
-
     if (getHash () == otherMap->getHash ())
         return true;
 
+    using StackEntry = std::pair <SHAMapTreeNode*, SHAMapTreeNode*>;
```

Comment:
These two definitions appears twice - you should hoist them up and have it "global" to this compilation unit.

    using SHAStackEntry = std::pair <SHAMapTreeNode*, SHAMapTreeNode*>;
    using SHAStack = std::stack <SHAStackEntry, std::vector<SHAStackEntry>>;